### PR TITLE
implemented augments_fasttext function

### PR DESCRIPTION
**Pull request by joeybodoia**

### CODE REVIEW

File: chajda/tsquery/augments.py [hunk 1]
```diff
@@ -46,10 +67,51 @@ def augments_gensim(lang, word, config=Config(), n=5):
...
+    except:
+        with suppress_stdout_stderr():
+            fasttext.util.download_model(lang, if_exists='ignore')
+        fasttext_models[lang] = fasttext.load_model('cc.{0}.300.bin'.format(lang))
+
+    #find the most similar words
+    try:
+        topn = fasttext_models[lang].get_nearest_neighbors(word, k=n)
+        words = ' '.join([ word for (rank, word) in topn ])
+    except KeyError:
```
Comment (mikeizbicki) [inline chajda/tsquery/augments.py]: What is the cause of the `KeyError`?

My guess is that this is caused by `fasttext_models[lang]` not containing the `lang` key.  This should only happen if the model couldn't be downloaded.  In that case, we should probably raise a `ValueError` to indicate that the specified language is incorrect.

File: chajda/tsquery/augments.py [hunk 2]
```diff
@@ -46,10 +67,51 @@ def augments_gensim(lang, word, config=Config(), n=5):
...
+    '(baby:A <1> boy:A) & ((school:A | schoo:B | schoolthe:B | schoool:B | kindergarten:B) | (home:A | house:B | homethe:B | homewhen:B | homethis:B)) & !(weapon:A | weaponthe:B | weopon:B)'
 
+    >>> augments_fasttext('en','weapon', n=5)
+    ['weaponthe', 'weopon']
+
+    >>> augments_fasttext('en','king', n=5)
+    ['queen', 'kingthe']
+    '''
+
+    try:
```
Comment (mikeizbicki) [inline chajda/tsquery/augments.py:94]: Add a comment describing what this "paragraph" is doing

File: chajda/tsquery/augments.py [hunk 3]
```diff
@@ -2,7 +2,20 @@
 '''
 
 from chajda.tsvector import lemmatize, Config
-from chajda.tsquery import to_tsquery
+from chajda.tsquery.__init__ import to_tsquery
+
+
+from contextlib import contextmanager,redirect_stderr,redirect_stdout
```
Comment (mikeizbicki) [inline chajda/tsquery/augments.py]: This chunk of code all comes from the stackoverflow question, right?

Whenever I copy/paste from stackoverflow, I always include a comment explaining why I've included the code here with a link to the original code.  These copy/pastes are often of esoteric concepts (as this one is), and so the comment can be extremely helpful in the future.

File: chajda/tsquery/augments.py [hunk 4]
```diff
@@ -46,10 +67,51 @@ def augments_gensim(lang, word, config=Config(), n=5):
...
+import fasttext.util
+
+fasttext_models = {}
+
+def augments_fasttext(lang, word, config=Config(), n=5):
+    ''' 
+    Returns n words that are "similar" to the input word in the target language.
+    These words can be used to augment a search with the Query class.
+
+    Note: Due to the size of fasttext models, testing multiple languages in the doctest requires more space than github actions allows for. For this reason, only english is tested in the doctest.
```
Comment (mikeizbicki) [inline chajda/tsquery/augments.py]: Reformat the note and add the es/ja test cases.

**Comment by mikeizbicki** [comment]
Thanks for squashing the commits.  In the future, you'll want to adjust the commit so that it doesn't include all of the old commit messages, but just a 1 line summary of everything together.

It's almost ready for a merge, but there's a few minor issues I'd still like to get fixed first.

**Comment by joeybodoia** [comment]
Thanks for helpful comments Mike, I'll review them and make the necessary changes.

File: chajda/tsquery/augments.py [hunk 5]
```diff
@@ -93,21 +94,37 @@ def augments_fasttext(lang, word, config=Config(), n=5):
...
+    topn = fasttext_models[lang].get_nearest_neighbors(word, k=n)
+    words = ' '.join([ word for (rank, word) in topn ])
+
     #find the most similar words
-    try:
-        topn = fasttext_models[lang].get_nearest_neighbors(word, k=n)
-        words = ' '.join([ word for (rank, word) in topn ])
-    except KeyError:
-        return []
+   # try:
```
Comment (mikeizbicki) [inline chajda/tsquery/augments.py]: Looks good except this block of commented out code should be deleted.

**Comment by mikeizbicki** [comment]
I still see 4 commits, so I don't think the squashing worked.  Overall, the contents of the commits look fine, but I'll let you try resquashing again before I merge just so you can get more practice with that workflow.

**Comment by joeybodoia** [comment]
Sorry about that, I forgot to use `git push --force`. Should be only one commit now.

**Comment by mikeizbicki** [comment]
great job!

**Pull request merged**